contracts/swapHandlers/UniswapV3SwapHandler.sol

```diff
@@ -0,0 +1,88 @@
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+pragma solidity ^0.8.0;
+
+import "./SwapHandlerBase.sol";
+import "../vendor/ISwapRouter.sol";
+
+/// @notice Swap handler executing trades on UniswapV3 through SwapRouter
+contract UniswapV3SwapHandler is SwapHandlerBase {
+    address immutable public uniswapRouter;
+
+    constructor(address uniswapRouter_) {
+        uniswapRouter = uniswapRouter_;
+    }
+
+    function executeSwap(SwapParams calldata params) override external {
+        setMaxAllowance(params.underlyingIn, uniswapRouter);
+
+        if (params.payload.length == 64) {
+            (uint sqrtPriceLimitX96, uint fee) = abi.decode(params.payload, (uint, uint));
+            if (params.mode == 0)
+                exactInputSingle(params, sqrtPriceLimitX96, fee);
+            else
+                exactOutputSingle(params, sqrtPriceLimitX96, fee);
+        } else {
+            if (params.mode == 0)
+                exactInput(params, params.payload);
+            else
+                exactOutput(params, params.payload);
+        }
+
+        if (params.mode == 1) transferBack(params.underlyingIn);
+    }
+
+    function exactInputSingle(SwapParams memory params, uint sqrtPriceLimitX96, uint fee) private {
+        ISwapRouter(uniswapRouter).exactInputSingle(
+            ISwapRouter.ExactInputSingleParams({
+                tokenIn: params.underlyingIn,
+                tokenOut: params.underlyingOut,
+                fee: uint24(fee),
+                recipient: msg.sender,
+                deadline: block.timestamp,
```

Comment:
if you use the newer router (SwapRouter02), the deadlines may be omitted altogether for all the functions

Reply:
Interesting, switching to 02 consistently adds ~3k gas overhead. Since there's no other benefit than not passing the deadline, I'd stick with the old router. In UniV3 handler only